src/components/ValueInput/ValueInput.js

```diff
@@ -220,6 +222,9 @@ export const ValueInputComponent = ({
   const openAssetSelector = () => {
     Keyboard.dismiss();
 
+    if (onValueChange) {
```

Comment:
I see two issue with this line:
1. When user opens asset selector, but then uses the back button to dismiss it the value is reset to zero.
2. It does not seem to work, when the ValueInput is in fiat input mode, opening the modal resets the token amount but not fiat value.

Consider clearing both token and fiat value when the selected token actually changes. That should address both of the issues above.

Reply:
I will look into this. Also let me know point 2, fiat input mode.